### PR TITLE
fix seed content registration bugs

### PR DESCRIPTION
**Pull request by DelusionalOptimist**
**Description**

This PR fixes #

**Notes for Reviewers**
* Panic due to meshkit error conversion
* Improves logic to read recursively through directories so that the new ServiceMeshCon patterns could be readed and registered too.
```
ERRO[0000] read /home/rudraksh/.meshery/seed_content/patterns/servicemeshcon-na-2021: is a directory
```

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: models/error.go [hunk 1]
```diff
@@ -280,3 +283,15 @@ func ErrFetchData(err error) error {
 func ErrSessionCopy(err error) error {
 	return errors.New(ErrSessionCopyCode, errors.Alert, []string{"Error: session copy error"}, []string{err.Error()}, []string{}, []string{})
 }
+
+func ErrSeedingPatterns(err error) error {
+	return errors.New(ErrSeedingPatternsCode, errors.Alert, []string{"Error while seeding patterns from seed content"}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrGettingSeededPatterns(err error) error {
+	return errors.New(ErrGettingSeededPatternsCode, errors.Alert, []string{"Error while reading seeded patterns."}, []string{err.Error()}, []string{}, []string{})
```
Comment (leecalcote) [inline models/error.go:292]: Can we add probable cause and suggested remediations?

Comment (DelusionalOptimist) [reply models/error.go:292]: sure

Comment (leecalcote) [reply models/error.go:292]: Probable cause: 

"Lack of Internet connectivity to download sample patterns, filters, and applications."

Comment (leecalcote) [reply models/error.go:292]: Probably cause: "Invalid sample content."

**Review by leecalcote — APPROVED** [review]
Nice!

**Comment by DelusionalOptimist** [comment]
Fixing failing backend build. Its due to a fucntion that was moved from a package to another.

**Pull request merged**